timelock-agent/src/main/java/com/palantir/timelock/paxos/PaxosAgent.java

```diff
@@ -0,0 +1,109 @@
+import com.palantir.remoting2.config.ssl.SslSocketFactories;
+import com.palantir.timelock.TimeLockAgent;
+import com.palantir.timelock.config.ImmutablePaxosRuntimeConfiguration;
+import com.palantir.timelock.config.ImmutableTimeLockDeprecatedConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration;
+import com.palantir.timelock.config.PaxosRuntimeConfiguration;
+import com.palantir.timelock.config.TimeLockDeprecatedConfiguration;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
+
+import io.reactivex.Observable;
+
+public class PaxosAgent extends TimeLockAgent {
+    private final PaxosInstallConfiguration paxosInstall;
+
+    private final PaxosResource paxosResource;
+    private final PaxosLeadershipCreator leadershipCreator;
+    private final LockCreator lockCreator;
+    private final PaxosTimestampCreator timestampCreator;
+    private final TimeLockServicesCreator timelockCreator;
+
+    public PaxosAgent(TimeLockInstallConfiguration install,
+            Observable<TimeLockRuntimeConfiguration> runtime,
+            Consumer<Object> registrar) {
+        this(install, runtime, ImmutableTimeLockDeprecatedConfiguration.builder().build(), registrar);
+    }
+
+    public PaxosAgent(TimeLockInstallConfiguration install,
+            Observable<TimeLockRuntimeConfiguration> runtime,
+            TimeLockDeprecatedConfiguration deprecated,
+            Consumer<Object> registrar) {
+        super(install, runtime, deprecated, registrar);
+        this.paxosInstall = install.algorithm();
+
+        this.paxosResource = PaxosResource.create(paxosInstall.dataDirectory().toString());
+        this.leadershipCreator = new PaxosLeadershipCreator(install, runtime, registrar);
+        this.lockCreator = new LockCreator(runtime, deprecated);
+        this.timestampCreator = new PaxosTimestampCreator(paxosResource,
+                PaxosRemotingUtils.getRemoteServerPaths(install),
+                PaxosRemotingUtils.getSslConfigurationOptional(install).map(SslSocketFactories::createSslSocketFactory),
+                runtime.map(PaxosAgent::getAlgorithmConfigOrDefault));
+        this.timelockCreator = install.asyncLock().useAsyncLockService()
+                ? new AsyncTimeLockServicesCreator(leadershipCreator, install.asyncLock())
+                : new LegacyTimeLockServicesCreator(leadershipCreator);
+    }
+
+    @Override
+    public void createAndRegisterResources() {
+        registerPaxosResource();
+        registerPaxosExceptionMappers();
+        leadershipCreator.registerLeaderElectionService();
+
+        // Finally, register the endpoints associated with the clients.
+        super.createAndRegisterResources();
+    }
+
+    // No runtime configuration at the moment.
+    private void registerPaxosResource() {
+        registrar.accept(paxosResource);
+    }
+
+    private void registerPaxosExceptionMappers() {
```

Comment:
Yep, will move this. Following @tpetracca's later comment I think it makes sense to converge the generic timelock agent and concrete paxos agent.